common/src/set-version-labels.js

```diff
@@ -0,0 +1,308 @@
+                return null;
+            }
+
+            console.debug(`Patch Versions to fetch first element:`, versionsJsonString);
+
+            const extractFirstPatchVersionFromJson = new RegExp(`(number\\: ')(.*)('\\,)`);
+            const matchFirstPatch = versionsJsonString.match(extractFirstPatchVersionFromJson);
+            let firstPatchVersion = matchFirstPatch ? matchFirstPatch[2] : null;
+
+            console.debug(`The latest Patch version is: ${firstPatchVersion}`);
+            return `version:` + firstPatchVersion;
+        } catch (error) {
+            console.error("Error while calculating Patch version:", error);
+            return null;
+        }
+    }
+
+    const getNextMinorVersion = async function() {
+        const url = `https://github.com/camunda/camunda-bpm-platform/raw/refs/heads/master/pom.xml`;
+        const response = await fetch(url);
+        const pomXml = await response.text();
+        const versionTagRegex = /<artifactId>camunda-root<\/artifactId>\s*<version>(\d+\.\d+)(?:\.\d+.*)?<\/version>/
+        const match = pomXml.match(versionTagRegex);
+        // Return the version if found, otherwise return null
+        return match ? `${match[1]}` : null;
+    }
+
+    const isNextMinorReleaseVersion = function (potentialLabel, nextMinorVersion) {
+        return getMinorFromPotentialLabel(potentialLabel) === nextMinorVersion
+    }
+
+    async function fetchDownloadPage() {
+        const url = `https://docs.camunda.org/enterprise/download/`;
+
+        return new Promise((resolve, reject) => {
+            https.get(url, (response) => {
+                let data = '';
+
+                if (response.statusCode !== 200) {
+                    reject(new Error(`Failed to fetch data. Status code: ${response.statusCode}`));
+                    response.resume(); // Consume response data to free up memory
+                    return;
+                }
+
+                response.on('data', (chunk) => {
+                    data += chunk;
+                });
+
+                response.on('end', () => {
+                    resolve(data);
+                });
+            }).on('error', (error) => {
+                reject(error);
+            });
+        });
+    }
+
+    const setLabels = async function (ticketMetadata, labels) {
+        console.log(`Set labels for issue: #${issueNumber}:`, labels);
+        try {
+            await octokit.rest.issues.addLabels({
+                ...ticketMetadata,
+                labels: labels
+            });
+
+            console.log(`Labels set to issue #${issueNumber}:`, labels);
+        } catch (error) {
+            console.error('Error setting labels:', error);
+        }
+    }
+
+    const postGithubComment = async (ticketMetadata, comment) => {
+        await octokit.rest.issues.createComment({
+            ...ticketMetadata,
+            body: comment,
+        });
+    }
+
+    const getCommentText = (potentialToVersionLabelsMap) => {
```

Comment:
🔧 If I understand it correctly, this function will only produce a comment in case of non-matched potentials and/or patch version mismatches. I would suggest to name the function `getErrorCommentText` or something else that reflects in which cases comments are posted.

Reply:
Honoured this point and renamed the function to `getWarningCommentText` 🙂